docs/core/advanced_tutorials/using-results.md

```diff
@@ -0,0 +1,226 @@
+
+@task
+def my_task():
+    # ...
+
+@task
+def my_second_task():
+    # ...
+
+with Flow("my handled flow!", result=gcs_result):
+    my_task()
+    my_second_task()
+```
+
+```bash
+$ gsutil ls -r gs://prefect_results
+gs://prefect_results/my_task.txt
+gs://prefect_results/my_second_task.txt
+```
+
+## Specifying a location for mapped or looped tasks
+
+Mapped or looped tasks require special consideration when it comes to their result location, as they may run many times and each of their results should be stored separately.
+
+## Running a flow with `PrefectResult`
+
+A unique case is the `PrefectResult`, as it serializes its `PrefectResult.value` as its `Result.location`. This is only useful for small data loads and for data that Cloud users are comfortable sharing with the Cloud database. When used effectively, they can provide efficient inspection of data output in the UI. For this reason, all `Parameter` type Tasks use this as their result subclass.
+
+Let's see this with an example. The same flow that adds numbers together will instead be configured with the `PrefectResult`:
+
+```python
+# flow.py
+from prefect.engine.result import PrefectResult
+
+with Flow("my handled flow!", result=PrefectResult()):
+    ...
+```
+
+Now when I run my flow, the `PrefectResult.location` contains the actual return value from my Task in it:
+
+```python
+>>> state.result[first_result]._result.value
+3
+>>> state.result[first_result]._result.location
+3
+```
+
+And this value `3` is also visible to Cloud users in the UI when inspecting the Task Run details:
+
+![Task Detail with JSON Result.safe_value showing](/result-stored-in-cloud-UI-jsonhandler.png)
+
+This is a useful backend for data that is small and safe to store in the database, as it requires no extra configuration.
+
+## Migrating from Result Handlers
+
+If you used result handlers in Prefect versions lower than 0.11.0, they have now been replaced by the `Result` subclasses described in this document.
+
+If you use result handlers in your flows and upgrade to 0.11.0, they will be auto-converted at runtime into `Result` subclasses matching the storage backend -- for example `GCSResultHandler`s will turn into `GCSResult`s. 
+
+Custom result handlers will be auto-converted at runtime into `ResultHandlerResult` classes, which act as a wrapper to expose the custom result handler's `read` and `write` methods.
+
+The conversion logic itself is located [in the code here](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/engine/results/result_handler_result.py).
+
+As you write new flows or upgrade existing flows, consider using the new `Result` subclasses wherever you would have used result handlers. We do not have an official timeline for removing result handlers entirely, but they are currently in maintenance mode and new features leveraging dataflow in Prefect will be designed and implemented against the `Result` interface only going forward.
```

Comment:
Is this true enough/clear enough?

Reply:
Yup this is definitely clear; because we convert ResultHandlers -> Results (and so aren't _really_ maintaining two code paths) we can probably hold onto result handlers for quite a while to let folks migrate